hal/src/rtl872x/exflash_hal.cpp

```diff
@@ -176,6 +392,99 @@ class ExFlashLock {
     bool threading_;
 };
 
+static int writeInBestModeWithinPageBoundaries(uint32_t addr, uint8_t* buf, size_t size) {
+    uint32_t address = addr;
+    uint8_t cmd = GENERIC_FLASH_CMD_PP;
+    uint32_t mode = RTK_SPIC_DATA_WIDTH_MODE_SINGLE;
+
+    if (flash_init_para.FLASH_cur_bitmode == SpicQuadBitMode) {
+        // Quad mode enabled
+        if (flash_init_para.FLASH_Id == FLASH_ID_GD) {
+            cmd = GD_FLASH_CMD_4PP;
+            mode = RTK_SPIC_DATA_WIDTH_MODE_QUAD_I;
+        } else {
+            cmd = MXIC_FLASH_CMD_4PP;
+            mode = RTK_SPIC_DATA_WIDTH_MODE_QUAD_II;
```

Comment:
Whats the difference between `RTK_SPIC_DATA_WIDTH_MODE_QUAD_I` and `RTK_SPIC_DATA_WIDTH_MODE_QUAD_II` ? 
Are these just differences between GD and MXIC protocols? These are both just saying the quad write command should have 3 byte addresses?

Reply:
`QUAD_II` is both 24-bit address and data sent over 4 IO lines, `QUAD_I` is 24-bit address over IO0 and data over 4 IO lines. Just slightly different modes. Just depends on what the flash supports.

![image](https://github.com/user-attachments/assets/75f0c1e8-e5f4-4a54-a59f-1b42c6fb4d28)
![image](https://github.com/user-attachments/assets/1601b5b7-b7df-4368-af00-fed0debeefac)